lib/config.go

```diff
@@ -549,10 +552,24 @@ func (c *NetworkConfig) Validate(onExecution bool) error {
 			return errors.Wrapf(err, "invalid network_dir: %s", err)
 		}
 
-		for _, pluginDir := range c.PluginDir {
+		for _, pluginDir := range c.PluginDirs {
 			if err := os.MkdirAll(pluginDir, 0755); err != nil {
+				return errors.Wrapf(err, "invalid plugin_dirs entry")
+			}
+		}
+		// While the plugin_dir option is being deprecated, we need this check
+		if c.PluginDir != "" {
+			logrus.Warnf("The config field plugin_dir is being deprecated. Please use plugin_dirs instead")
+			if err := os.MkdirAll(c.PluginDir, 0755); err != nil {
 				return errors.Wrapf(err, "invalid plugin_dir entry")
 			}
+			// Append PluginDir to PluginDirs, so from now on we can operate in terms of PluginDirs and not worry
+			// about missing cases.
+			c.PluginDirs = append(c.PluginDirs, c.PluginDir)
+
+			// Empty the pluginDir so on future config calls we don't print it out
+			// thus seemlessly transitioning and depreciating the option
+			c.PluginDir = ""
```

Comment:
Reviewers: note this change vs the 1.14 version. This means that 1.15 will phase out plugin_dir all together, and 1.16 can drop support.